python/lsst/obs/base/_instrument.py

```diff
@@ -398,6 +398,27 @@ def makeDataIdTranslatorFactory(self) -> TranslatorFactory:
         """
         raise NotImplementedError("Must be implemented by derived classes.")
 
+    @classmethod
+    def constructDefaultCollectionName(cls, label: str) -> str:
+        """Get the default instrument-specific collection string from the
+        supplied label.
+
+        Parameters
+        ----------
+        label : `str`
+            Generic type of collection from which to derive the default
+            name.  For example for ``raw`` this would return a string of the
+            form ``instrument_name/raw/all``.
+
+        Returns
+        -------
+        coll : `str`
+            Collection name to be used as the default for this generic label.
+        """
+        if label == "raw":
+            label = "raw/all"
+        return cls.constructCollectionName(label)
+
```

Comment:
I'm not sure see the point of this function, unless "label" really means "dataset type name" (which it does in at least some usage contexts below; maybe all), and the intent is to return the collection that dataset type should be ingested into, not retrieved from (certainly if we have `<instrument>/raw/default`, this _isn't_ the the collection the dataset should be retrieved from by default, so calling it "default" is going to be confusing).

I'm not sure if the solution is to make it a function that _is_ just that (a dataset type name -> ingest collection mapping), or if it's better to just make it a raw-specific function with no arguments.

Reply:
I'm happy to change it to a straight up makeDefaultRawCollection method for now. I was worried about having to go in and put "raw/all" everywhere and then have us change our mind and have to go through all the code again. Given that we haven't sorted out the calib API properly that might be best (and calib is not a dataset type).